open-budget/app/scripts/url_scheme.coffee

```diff
@@ -0,0 +1,93 @@
+window.DEFAULT_HOME = "#main//2014/main"
+class URLSchemeHandler
+
+        constructor: (pageModel) ->
+            @oldHash = window.location.hash
+            @callbackList = []
+            @reload = true
+            @pageModel = pageModel
+
+            @parseLink()
+
+            window.onhashchange = @handleSchemeChange
+
+
+        linkToBudget: (code,year) ->
+            @linkParameters['type'] = 'budget'
+            @linkParameters['code'] = code
+            @linkParameters['year'] = year
+            @buildLink()
+
+        linkToTransfer: (code,year) ->
+            @linkParameters['type'] = 'transfer'
+            @linkParameters['code'] = code
+            @linkParameters['year'] = year
+            @buildLink()
+
+        linkToEntity: (entityId) ->
+            @linkParameters['type'] = 'entity'
+            @buildLink()
+
+        addAttribute: (key, value, reload) ->
+            @linkParameters['attributes'][key] = value
+            @reload = reload || false
+            window.location.hash = @buildLink()
+
+        getAttribute: (key) ->
+            @linkParameters['attributes'][key]
+
+        buildLink: () ->
+            link = ""
+            switch @linkParameters['type']
+                when 'budget' then link = "#budget/#{@linkParameters['code'].slice(2)}/#{@linkParameters['year']}/#{@pageModel.get('flow')}"
+                when 'transfer' then link = "#transfer/#{@linkParameters['code']}/#{@linkParameters['year']}/#{@pageModel.get('flow')}"
+                when 'entity' then link = "#entity/#{@linkParameters['entityId']}/#{@pageModel.get('year')}/#{@pageModel.get('flow')}"
+
+            if Object.keys(@linkParameters['attributes']).length > 0
+                link += "?"+$.param(@linkParameters['attributes'])
+
+            return link
+
+        parseLink: () ->
+            @linkParameters = {
+                attributes: {}
+            }
+
+            hash = window.location.hash.substring(1)
+            [hashPath, attributeString] = hash.split("?")
+            [kind, identifier, year, flow] = hashPath.split("/",4)
+            year = parseInt(year)
+
+            console.log "hash:", kind, identifier, year, flow
+
+            @linkParameters['kind'] = kind
+            switch @linkParameters['kind']
+                when 'budget', \
+                     'transfer', \
+                     'main'
+                    @linkParameters['code'] = identifier
+                when 'entity'
+                    @linkParameters['entityId'] = identifier
+
+            @linkParameters['year'] = year || new Date().getFullYear()
+            @linkParameters['flow'] = flow
+            if attributeString && attributeString.length > 0
+                @linkParameters['attributes'] = JSON.parse('{"' + decodeURI(attributeString.replace(/&/g, "\",\"").replace(/\=/g,"\":\"")) + '"}')
+
+            @linkParameters
+
+
+        onSchemeChange: (callback) ->
+            @influencerList.push(callback)
```

Comment:
No, I ended up using a different mechanism.

Nevertheless, the logic requires an AND condition because the idea is that you can supply a callback that will allow you to cancel the reload for certain hash changes.